Content.Server/_DV/Abilities/Kitsune/KitsuneSystem.cs

```diff
@@ -0,0 +1,74 @@
+using Content.Server.Access.Systems;
+using Content.Server.Actions;
+using Content.Server.Polymorph.Systems;
+using Content.Server.Popups;
+using Content.Shared._DV.Abilities.Kitsune;
+using Content.Shared.Access.Components;
+using Content.Shared.Access.Systems;
+using Content.Shared.NPC.Components;
+using Content.Shared.NPC.Systems;
+using Content.Shared.Polymorph;
+using Robust.Shared.Player;
+
+namespace Content.Server._DV.Abilities.Kitsune;
+public sealed class KitsuneSystem : SharedKitsuneSystem
+{
+    [Dependency] private readonly SharedAppearanceSystem _appearance = default!;
+    [Dependency] private readonly PolymorphSystem _polymorph = default!;
+    [Dependency] private readonly PopupSystem _popup = default!;
+    [Dependency] private readonly AccessSystem _access = default!;
+    [Dependency] private readonly AccessReaderSystem _reader = default!;
+    [Dependency] private readonly NpcFactionSystem _faction = default!;
+    [Dependency] private readonly ActionsSystem _actions = default!;
+
+    public override void Initialize()
+    {
+        base.Initialize();
+        SubscribeLocalEvent<KitsuneComponent, MorphIntoKitsune>(OnMorphIntoKitsune);
+        SubscribeLocalEvent<KitsuneComponent, PolymorphedEvent>(OnPolymorphed);
+    }
+
+    private void OnPolymorphed(Entity<KitsuneComponent> ent, ref PolymorphedEvent args)
+    {
+        _appearance.SetData(args.NewEntity, KitsuneColorVisuals.Color, ent.Comp.Color ?? Color.Orange);
+
+        // Ensure that the fox fire action state is transferred properly.
+        if (!TryComp<KitsuneComponent>(args.NewEntity, out var newKitsune)
+            || !TryComp<KitsuneComponent>(ent, out var oldKitsune))
+            return;
+        newKitsune.ActiveFoxFires = oldKitsune.ActiveFoxFires;
+
+        _actions.SetCharges(newKitsune.FoxfireAction, _actions.GetCharges(oldKitsune.FoxfireAction));
+
+        foreach (var foxFire in newKitsune.ActiveFoxFires)
+        {
+            if(!TryComp<FoxfireComponent>(foxFire, out var foxfire))
+                continue;
+            foxfire.Kitsune = args.NewEntity;
+        }
```

Comment:
rename foxFire to fireUid and add `Dirty(fireUid, foxfire);` after setting Kitsune